worker version check failure crashes server

If the worker version doesn't match the server it is supposed to error out and the server should handle that gracefully and report it in the task status.  Right now we get this on the server:
```
Traceback (most recent call last):
  File "/usr/local/bin/turbiniactl", line 11, in <module>
    load_entry_point('turbinia==20181004', 'console_scripts', 'turbiniactl')()
  File "/usr/local/lib/python2.7/dist-packages/turbinia/turbiniactl.py", line 467, in main
    server.start()
  File "/usr/local/lib/python2.7/dist-packages/turbinia/client.py", line 766, in start
    self.task_manager.run()
  File "/usr/local/lib/python2.7/dist-packages/turbinia/task_manager.py", line 281, in run
    for task in self.process_tasks():
  File "/usr/local/lib/python2.7/dist-packages/turbinia/task_manager.py", line 426, in process_tasks
    task.stub.result(timeout=PSQ_TASK_TIMEOUT_SECONDS))
  File "/usr/local/lib/python2.7/dist-packages/turbinia/workers/__init__.py", line 287, in deserialize
    result.__dict__.update(input_dict)
TypeError: 'TurbiniaTaskResult' object is not iterable
```
And the worker:
[ERROR] Worker and Server versions do not match: 20181004 != 20190819

Also need to double check that the post-processors run properly to clean things up.

--- a/turbinia/workers/__init__.py
+++ b/turbinia/workers/__init__.py
@@ -613,7 +613,7 @@ class TurbiniaTask(object):
                   self.turbinia_version, turbinia.__version__))
           self.result.log(message, level=logging.ERROR)
           self.result.status = message
-          return self.result
+          return self.result.serialize()
 
         self._evidence_config = evidence.config
         self.result = self.run(evidence, self.result)

